libs/langchain/tests/integration_tests/vectorstores/test_bagel.py

```diff
@@ -0,0 +1,187 @@
+        texts=texts,
+        metadatas=metadatas,
+        embedding=embedding_function,
+    )
+    embedded_query = embedding_function.embed_query("hello bagel")
+    output = vector_search.similarity_search_by_vector_with_relevance_scores(
+        query_embeddings=embedded_query, k=1
+    )
+    assert output == [
+        (Document(page_content="hello bagel", metadata={"page": "0"}), 0.0)
+    ]
+    vector_search.delete_cluster()
+
+
+def test_search_filter() -> None:
+    """Test end to end construction and search with metadata filtering."""
+    texts = ["hello bagel", "hello langchain"]
+    metadatas = [{"first_letter": text[0]} for text in texts]
+    txt_search = Bagel.from_texts(
+        cluster_name="testing",
+        texts=texts,
+        metadatas=metadatas,
+    )
+    output = txt_search.similarity_search("bagel", k=1, where={"first_letter": "h"})
+    assert output == [
+        Document(page_content="hello bagel", metadata={"first_letter": "h"})
+    ]
+    output = txt_search.similarity_search("langchain", k=1, where={"first_letter": "h"})
+    assert output == [
+        Document(page_content="hello langchain", metadata={"first_letter": "h"})
+    ]
+    txt_search.delete_cluster()
+
+
+def test_search_filter_with_scores() -> None:
+    texts = ["hello bagel", "this is langchain"]
+    metadatas = [{"source": "notion"}, {"source": "google"}]
+    txt_search = Bagel.from_texts(
+        cluster_name="testing",
+        texts=texts,
+        metadatas=metadatas,
+    )
+    output = txt_search.similarity_search_with_score(
+        "hello bagel", k=1, where={"source": "notion"}
+    )
+    assert output == [
+        (Document(page_content="hello bagel", metadata={"source": "notion"}), 0.0)
+    ]
+    txt_search.delete_cluster()
+
+
+def test_with_include_parameter() -> None:
+    """Test end to end construction and include parameter."""
+    texts = ["hello bagel", "this is langchain"]
+    docsearch = Bagel.from_texts(cluster_name="testing", texts=texts)
+    output = docsearch.get(include=["embeddings"])
+    assert output["embeddings"] is not None
+    output = docsearch.get()
+    assert output["embeddings"] is None
+    docsearch.delete_cluster()
+
+
+def test_bagel_update_document() -> None:
+    """Test the update_document function in the Bagel class."""
+    initial_content = "bagel"
+    document_id = "doc1"
+    original_doc = Document(page_content=initial_content, metadata={"page": "0"})
+
+    docsearch = Bagel.from_documents(
+        cluster_name="testing_docs",
+        documents=[original_doc],
+        ids=[document_id],
+    )
+
+    updated_content = "updated bagel doc"
+    updated_doc = Document(page_content=updated_content, metadata={"page": "0"})
+    docsearch.update_document(document_id=document_id, document=updated_doc)
+    output = docsearch.similarity_search(updated_content, k=1)
+    assert output == [Document(page_content=updated_content, metadata={"page": "0"})]
+
+
+def main() -> None:
```

Comment:
I'm not sure L:172-L:187 code should be here. Isn't it a part of development that was not removed?

Reply:
Yeah. Fixing it.